logging.go

```diff
@@ -147,3 +151,140 @@ func (lr *LoggingResource) ValidateAWSTrustPolicy(ctx context.Context, awsExtern
+// Return an error to stop processing and bubble up the error.
+type NetworkFlowLogHandler func(log NetworkFlowLog) error
+
+// GetNetworkFlowLogs streams network flow logs for the tailnet, calling the provided
+// handler function for each log entry as it's parsed from the JSON response.
+// This approach is memory-efficient and handles large datasets without loading all logs into memory.
+//
+// Both start and end parameters are required by the server.
+// Times older than 30 days will be automatically adjusted by the server to the retention limit.
+func (lr *LoggingResource) GetNetworkFlowLogs(ctx context.Context, params NetworkFlowLogsRequest, handler NetworkFlowLogHandler) error {
+
+	url := lr.buildTailnetURL("logging", "network")
+	q := url.Query()
+	q.Set("start", params.Start.Format(time.RFC3339))
+	q.Set("end", params.End.Format(time.RFC3339))
+	url.RawQuery = q.Encode()
+
+	req, err := lr.buildRequest(ctx, http.MethodGet, url)
+	if err != nil {
+		return err
+	}
+
+	return lr.streamNetworkFlowLogs(req, handler)
+}
+
+// streamNetworkFlowLogs performs the streaming JSON parsing of network flow logs
+func (lr *LoggingResource) streamNetworkFlowLogs(req *http.Request, handler NetworkFlowLogHandler) error {
+	lr.init()
+	resp, err := lr.HTTP.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("HTTP %d: %s", resp.StatusCode, string(body))
+	}
+
+	decoder := json.NewDecoder(resp.Body)
+
+	token, err := decoder.Token()
+	if err != nil {
+		return fmt.Errorf("failed to read opening brace: %w", err)
+	}
+	if delim, ok := token.(json.Delim); !ok || delim != '{' {
+		return fmt.Errorf("expected opening brace, got %v", token)
+	}
+
+	token, err = decoder.Token()
+	if err != nil {
+		return fmt.Errorf("failed to read field name: %w", err)
+	}
+	if fieldName, ok := token.(string); !ok || fieldName != "logs" {
+		return fmt.Errorf("expected 'logs' field, got %v", token)
+	}
+
+	token, err = decoder.Token()
+	if err != nil {
+		return fmt.Errorf("failed to read logs array start: %w", err)
+	}
+	if delim, ok := token.(json.Delim); !ok || delim != '[' {
+		return fmt.Errorf("expected opening bracket for logs array, got %v", token)
+	}
+
+	for decoder.More() {
+		// Check if context is cancelled during streaming
+		select {
+		case <-req.Context().Done():
+			return req.Context().Err()
+		default:
+		}
+
+		var log NetworkFlowLog
+		if err := decoder.Decode(&log); err != nil {
+			return fmt.Errorf("failed to decode log entry: %w", err)
+		}
+
+		if err := handler(log); err != nil {
+			return fmt.Errorf("handler error: %w", err)
+		}
+	}
+
+	token, err = decoder.Token()
+	if err != nil {
+		return fmt.Errorf("failed to read logs array end: %w", err)
+	}
+	if delim, ok := token.(json.Delim); !ok || delim != ']' {
+		return fmt.Errorf("expected closing bracket for logs array, got %v", token)
+	}
```

Comment:
We do this several times, it might be worth a little helper. (Example untested)

```suggestion
func checkDelim(dec *json.Decoder, want rune) error {
	token, err = decToken()
	if err != nil {
		return fmt.Errorf("failed to read delimiter: %w", err)
	}
	if delim, ok := token.(json.Delim); !ok || delim != want {
		return fmt.Errorf("got token %v, want %c", token, want)
	}
	return nil
}

// then...
if err := checkDelim(decoder, ']'); err != nil{
   return err
}
```

for a bit less ceremony at the usage. YMMV, up to you.